revoke/revoke.go

```diff
@@ -123,69 +154,145 @@ func getIssuer(cert *x509.Certificate) *x509.Certificate {
 	}
 
 	return issuer
-
 }
 
-// check a cert against a specific CRL. Returns the same bool pair
-// as revCheck.
-func certIsRevokedCRL(cert *x509.Certificate, url string) (revoked, ok bool) {
-	crl, ok := CRLSet[url]
+// checks whether CRL in memory is valid
+func (r *Revoke) isInMemoryCRLValid(key string) bool {
+	crl, ok := r.CRLSet[key]
 	if ok && crl == nil {
 		ok = false
-		delete(CRLSet, url)
+		r.CRLSetLck.Lock()
```

Comment:
this actually unsafe operation.  One thread will do "r.CRLSet[key]" read and the other will do delete. Only locking on delete doesn't prevent the read. I suggest we don't do locking for now and focusing on functionality.